nntrainer/src/tensor.cpp

```diff
@@ -783,17 +783,24 @@ Tensor Tensor::apply(std::function<float(float)> f) const {
 Tensor Tensor::apply(std::function<Tensor(Tensor)> f) const { return f(*this); }
 
 void Tensor::print(std::ostream &out) const {
-  unsigned int i, j, k, l;
-  std::stringstream ss;
-
+  out << "<Tensor Object at " << this << ">" << std::endl;
   const float *data = getData();
+
+  unsigned int len = length();
+
+  if (len > 100) {
+    out << '[' << data[0] << ' ' << data[1] << ' ' << data[2] << " ... "
+        << data[len - 3] << ' ' << data[len - 2] << ' ' << data[len - 1] << ']'
+        << std::endl;
+    return;
+  }
+
+  unsigned int i, j, k, l;
```

Comment:
1. If these iterator variables are used only for the following 'for' loops, you can move them into corresponding 'for' statements.

2. Comments for the following 'for' statements: 
dim.batch(), dim.channel(), dim.height(), dim.width(), and dim.getFeatureLen() would be invoked multiple times for no reasons. Please use variables for them outside the following 'for' loops.

Reply:
Q. Regrading 2. those are inline functions. Does using a variable outside of the loop make a difference?

Reply:
Updated for 1. Thanks :)

Reply:
A for 2. https://isocpp.org/wiki/faq/inline-functions#inline-and-perf

Reply:
Thanks you for the explanation :)